### PR TITLE
remove unneeded async fns, remove esModuleInterop, upgrade output to es6, bump package.json version to v2

### PR DESCRIPTION
**Pull request by ryanio**
Closes https://github.com/ProjectOpenSea/seaport-js/pull/128

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,11 +1,10 @@
 {
   "compilerOptions": {
     "outDir": "lib",
-    "target": "es5",
+    "target": "es6",
```
Comment (JoshuaSchmidt-OpenSea) [inline tsconfig.json:4]: Any desire to go further than this? Bigint was added in ES2020 but I don't know the downside of using a fairly new compiler target

Comment (ryanio) [reply tsconfig.json:4]: i think going to es6 is fine for now. we can do es2020 with ethers v6 + bigint upgrade

**Review by JoshuaSchmidt-OpenSea — APPROVED** [review]
Is the target version bump the reason for the major version upgrade?

**Comment by ryanio** [comment]
actually the initial reason was changing the async signatures on some fns:
- fulfillBasicOrder in src/utils/fulfill.ts
- fulfillStandardOrder in src/utils/fulfill.ts
- getApprovalActions in src/utils/approval.ts

**Pull request merged**